pyscf/adc/radc.py

```diff
@@ -460,8 +523,25 @@ def kernel(self, nroots=1, guess=None, eris=None):
             self.check_sanity()
         self.dump_flags_gs()
     
-        eris = radc_ao2mo.transform_integrals_incore(self)
-        self.e_corr, self.t1, self.t2 = compute_amplitudes_energy(self, eris, verbose=self.verbose)
+        nmo = self._nmo
+        nao = self.mo_coeff.shape[0]
+        nmo_pair = nmo * (nmo+1) // 2
+        nao_pair = nao * (nao+1) // 2
+        mem_incore = (max(nao_pair**2, nmo**4) + nmo_pair**2) * 8/1e6
+        mem_now = lib.current_memory()[0]
+
+        if (self._scf._eri is not None and
+            (mem_incore+mem_now < self.max_memory or self.incore_complete)):
+            eris = self.transform_integrals()
```

Comment:
eris = self.transform_integrals() should be moved out of if and else statement below, since it's repeated.